### DISCUSSION TITLE
Python Programlama | SonsuzUs Github

### DISCUSSION THREAD

**Discussion by giscus** [sonsuzus/sonsuzus.github.io#112] [Site]
# Python Programlama | SonsuzUs Github

Bu bölümde python programlama ile ilgili dersler ve ipuçları olacaktır. Aşama aşama konular ve açıklamları ayrıca önemli ipuçlarının bulunacağı bu bölümle ilgili sorularınızı yorum bölümünden sorabilirsiniz.

https://sonsuzus.github.io/posts/python/

**Comment by sonsuzus** [comment]
### Python hakkında

Python genel amaçlı, yüksek seviyeli, yorumlanan, dinamik algılama yapısına sahip ve nesneye dayalı olarak kullanılabilen bir programlama dilidir.

1991 yılında yayınlanan Python, Guido Van Rossum tarafından geliştirilmiştir.

Python'da yazılan programların çalışabilmeleri için kaynak kodlarının bir yorumlayıcı tarafından yorumlanması gerektiğinden, programın çalışacağı her bilgisayara ayrıca yorumlayıcı bir program yüklenmelidir.

Dinamik algılama yapısı ile bir değişken tanımlandığında, değişkene atanan değerin veri türü otomatik olarak algılanır ve değişkenin veri türü bu değerin veri türüne göre belirlenir.

### Python özellikleri

Python Program Dili'nin bu kadar yaygın olarak kullanılmasının başlıca nedenleri aşağıda gösterilmektedir:

- Öğrenmesi oldukça kolaydır.
- Ücretsiz ve açık kaynak kodlu bir programlama dilidir.
- Daha az kod kullanarak yazılım geliştirilebilir.
- Veri bilimi, makine öğrenimi ve yapay zeka alanlarında yaygın olarak kullanılır.
- Nesneye yönelik uygulamalar geliştirilebilir.
- Platformdan bağımsız uygulamalar geliştirilmesine olanak sağlar. Aynı kod ile geliştirilen bir uygulama Windows, Mac ve Linux işletim sistemlerine sahip bilgisayarlarda çalıştırılabilir.
- Python programını destekleyen bir çok kütüphane vardır.
- Bir çok programlama dili ile entegre edilebilir.

### Uygulama türleri

Python Program Dili çok farklı alanlarda kullanılmaktadır. Bu alanların başlıcaları aşağıda gösterilmektedir:

- Yazılım geliştirme
- Web uygulamaları
- Veri analizi
- Makine öğrenimi
- Oyun geliştirme
- Otomasyon
